packages/expect/src/matchers.ts

```diff
@@ -368,107 +372,118 @@ const matchers: MatchersObject = {
 
   toContain(
     this: MatcherState,
-    collection: ContainIterable | string,
-    value: unknown,
+    received: ContainIterable | string,
+    expected: unknown,
   ) {
-    const collectionType = getType(collection);
+    const options: MatcherHintOptions = {
+      comment: 'indexOf',
+      isNot: this.isNot,
+      promise: this.promise,
+    };
 
-    let converted: any = null;
-    if (Array.isArray(collection) || typeof collection === 'string') {
-      // strings have `indexOf` so we don't need to convert
-      // arrays have `indexOf` and we don't want to make a copy
-      converted = collection;
-    } else {
-      try {
-        converted = Array.from(collection);
-      } catch (e) {
-        throw new Error(
-          matcherErrorMessage(
-            matcherHint('.toContain', undefined, undefined, {
-              isNot: this.isNot,
-            }),
-            `${RECEIVED_COLOR(
-              'received',
-            )} value must not be null nor undefined`,
-            printWithType('Received', collection, printReceived),
-          ),
-        );
-      }
+    if (received === null || received === undefined) {
+      throw new Error(
+        matcherErrorMessage(
+          matcherHint('toContain', undefined, undefined, options),
+          `${RECEIVED_COLOR('received')} value must not be null nor undefined`,
+          printWithType('Received', received, printReceived),
+        ),
+      );
     }
+
+    const isArray = Array.isArray(received); // has indexOf
+    const isString = typeof received === 'string'; // has indexOf
+    const converted: any =
```

Comment:
I know this was already `any` previously, but wouldn't `string | Array<unknown>` be possible and at least type the `indexOf` and allow us to remove the comment below?

Reply:
Yeah, what do you think if instead of disjunction type, I separate the code paths:
* string
* array/iterable and try to factor out common code for `.toContain` and `.toContainEqual`